05/app.js

```diff
@@ -0,0 +1,96 @@
+function Student (initFirsName, initLastName) {
+    this.firstName = initFirsName;
+    this.lastName = initLastName;
+    this.grades = {
+
+    }
+}
+
+Student.prototype.showFullName = function() {
+    console.log(this.firstName, this.lastName);
+}
+
+Student.prototype.addGrade = function(subject, grade) {
+
+    if (typeof this.grades[subject] === 'undefined') {
+        this.grades[subject] = [];
+    }
+    this.grades[subject].push(grade);
+}
+
+Student.prototype.showGrades = function() {
+    // console.log(this.grades);
+    for (const element in this.grades) {
+        console.log(element + ':');
+        console.log(this.grades[element]);
+    }
+}
+
+Student.prototype.getAverageGrade = function(subject) {
+
+    if (subject !== undefined) {
+
+        let oneGradeTotal = 0;
+        let numberGrades = 0;
+        for (const el in this.grades) {
+
+            if (subject === el) {
+
```

Comment:
Łatwiej jest to zrealizować przez `if(typeof this.grades[subject] !== 'undefined') { ... }`